cpp/src_prims/vectorized.cuh

```diff
@@ -24,6 +24,27 @@ namespace MLCommon {
 template <typename math_, int VecLen>
 struct IOType {};
 
+template <>
+struct IOType<bool, 1> {
+  typedef int8_t Type;
```

Comment:
The size of bool is not defined by the standard, therefore I am not 100% confident that this is the best way.

This is needed to enable binaryOp to return a bool flag by comparing float values (used in Results::SelectUnboundSV).

Reply:
good point. maybe a `static_assert` to check `sizeof(bool) == sizeof(int8_t)` would be better?

Reply:
Yes, that is a good idea, I have added it.